File: src/types/order.ts
// src/types/order.ts
export interface OrderItemDto {
    id: number;
    productId: number;
    productName: string;
    quantity: number;
    unitPrice: number;
    totalPrice: number;
}

export interface OrderDto {
    id: number;
    userId: number;
    userName: string;
    orderDate: string;
    status: string;
    totalAmount: number;
    shippingAddress: string;
    paymentMethod: string;
    orderItems: OrderItemDto[];
}

export interface CreateOrderItemDto {
    productId: number;
    quantity: number;
}

export interface CreateOrderDto {
    shippingAddress: string;
    paymentMethod: string;
    orderItems: CreateOrderItemDto[];
}

export interface UpdateOrderStatusDto {
    status: string;
}

File: ECommerceApp.Domain/Entities/Order.cs
using ECommerceApp.Domain.Common;

namespace ECommerceApp.Domain.Entities
{
    public class Order : BaseAuditableEntity
    {
        public int UserId { get; set; }
        public User User { get; set; } = null!;
        public DateTime OrderDate { get; set; }
        public string Status { get; set; } = string.Empty;
        public decimal TotalAmount { get; set; }
        public string ShippingAddress { get; set; } = string.Empty;
        public string PaymentMethod { get; set; } = string.Empty;
        public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
    }
}
